### PR TITLE
Add "deprecation" docs with screenshots

### LINKED ISSUE #72: README doesn't explain the rule
I read the README extensively and it was difficult for me to understand what this plugin does.

At the beginning it says that's a rule for deprecated code, but it's not easy to understand what's deprecated code.

The rest is about installation and set up.

I ended up looking at the tests.

Maybe before the installation header, the documentation should show **an example** of what the rule does.

### PR DESCRIPTION
**Pull request by karlhorky**
Closes #72

Keeps the wording very short, but gives a bunch of examples with code blocks:

Screenshot 2023-10-15 at 13 10 09

### CODE REVIEW

**Comment by karlhorky** [comment]
Alternative screenshot here (using the Error Lens VS Code Extension to show multiple errors at the same time):

Screenshot 2023-10-11 at 23 31 10

**Comment by jenssimon** [comment]
I totally agree with this PR. But I really don't like the screen shots.

We all love our editor configuration. It looks nice. But it's obstrusive.

Using code blocks to explain the rule is unopionionated. And usage of emojis is fine.

File: README.md [hunk 1]
```diff
@@ -11,6 +11,20 @@
 
 > An [ESLint](https://eslint.org/) rule that reports usage of deprecated code.
 
+Reports usage of anything marked with a `@deprecated` JSDoc tag (browser APIs, Node.js APIs, library APIs, etc)
```
Comment (gund) [inline README.md:14]:
```suggestion
Reports usage of any deprecated code which is marked with a [`@deprecated` JSDoc tag](https://jsdoc.app/tags-deprecated.html).

```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,20 @@
 
 > An [ESLint](https://eslint.org/) rule that reports usage of deprecated code.
 
+Reports usage of anything marked with a `@deprecated` JSDoc tag (browser APIs, Node.js APIs, library APIs, etc)
+
+**Deprecated browser APIs**
+
+<img width="620" src="https://github.com/gund/eslint-plugin-deprecation/assets/1935696/361d6546-aa74-4b32-b977-36b6e3f183fa" alt="Screenshot of warning `'event' is deprecated` raised by eslint-plugin-deprecation" /><br />
```
Comment (gund) [inline README.md:18]: We should not use screenshots, instead markdown already supports code snippets, you can annotate even the language used in a snippet, for example TS would be marked as \```ts code here```.

I would suggest removing extra examples as they do not provide much value, from just one example you can easily explain what this rule does.

Also as I mentioned in the issue this explanation docs should come after prerequisites and install section because that is the main purpose of the docs.

**Review by gund — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (gund) [reply README.md:18]: We can use similar structure for examples by showing incorrect/correct versions of the code, for example like [this rule](https://eslint.org/docs/latest/rules/no-await-in-loop).

File: README.md (hunk 1 above)
Comment (karlhorky) [reply README.md:14]: I think this removes a main point of value for this PR (clarifying the unclear point of "which code could this be?" by removing the 3 examples)

File: README.md (hunk 2 above)
Comment (karlhorky) [reply README.md:18]: > We should not use screenshots, instead markdown already supports code snippets, you can annotate even the language used in a snippet, for example TS would be marked as ```ts code here```.

Eh, I think it makes it less tangible like this, but...

> We can use similar structure for examples by showing incorrect/correct versions of the code, for example like this rule.

...Since this is a common pattern in other ESLint plugins, I will review the common patterns and follow that one

Comment (karlhorky) [reply README.md:18]: > I would suggest removing extra examples as they do not provide much value, from just one example you can easily explain what this rule does.

Strong disagree, just like the 3 examples of "what types of code could this be?" in the first paragraph, it's important that there are multiple code examples. Not sure if you saw the confusion on Twitter from multiple users, but it's pretty clear people find it hard to understand what this plugin actually is for.

Comment (gund) [reply README.md:18]: 2 examples should be fine - one for incorrect and one for correct cases and this is similar to what most eslint rules do.

File: README.md (hunk 1 above)
Comment (gund) [reply README.md:14]: I specifically do not want to mention any APIs whatsoever and added "any" to "deprecated code" phrase to substitute specifics, otherwise this may lead to more confusion, such as why specifically those APIs are mentioned, does that mean it will not work for other use-cases etc.
This rule is fully agnostic and does not care what kinds of APIs you are using or where the code is coming from, as long as it's JS/TS code which is marked with deprecated jsdoc it will report it's usage.

File: README.md (hunk 2 above)
Comment (karlhorky) [reply README.md:18]: I will take a look at what other plugins do and what your plugin does specially (it surfaces the deprecation messages, which is a unique feature and should be reflected in the documentation), but I'm 👎 on reducing the number of examples. Actually there should be more than 3.

Comment (karlhorky) [reply README.md:18]: To see how deficient your documentation is, just check out a well-documented plugin like `eslint-plugin-unicorn`:

- main docs page: https://github.com/sindresorhus/eslint-plugin-unicorn
- separate rule documentation page (lots of examples): https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat.md

File: README.md (hunk 1 above)
Comment (karlhorky) [reply README.md:14]: I think you're misunderstanding the point of providing these examples, so let me try again:

People all over the place are not understanding what `eslint-plugin-deprecation` does. They don't know what type of code it could possibly apply to. Saying "any code" is technically, purely correct, but completely dismisses the difficulties of these people and lacks proper teaching (which is what docs should do)

Teaching can be easily done here by providing some simple examples of what types of code this plugin deals with. Examples are a pretty common teaching tool. These examples have already proved to be valuable by my responses on Twitter to users confused by the current docs (see #72).

Adding examples of what types of code could apply here (including "etc." or similar language) does not discount from the words "any code" in any way, and does not indicate that `eslint-plugin-deprecation` does not handle other types of code outside of the examples

Comment (karlhorky) [reply README.md]: I updated "anything" to "any code" and added "incl."

```diff
-Reports usage of anything marked with a `@deprecated` JSDoc tag (browser APIs, Node.js APIs, library APIs, etc)
+Reports usage of any code marked with a `@deprecated` JSDoc tag (incl. browser APIs, Node.js APIs, library APIs, etc.)
```

File: README.md (hunk 2 above)
Comment (karlhorky) [reply README.md]: @gund as a followup to this, you may want to consider integrating `eslint-doc-generator` to generate a documentation page for the `deprecated` rule, as demonstrated in this PR:

- https://github.com/eslint-community/eslint-plugin-security/pull/101

**Comment by karlhorky** [comment]
@gund @jenssimon I've switched from screenshots to code blocks, following the patterns of official ESLint rules docs (incorrect/correct examples):

Screenshot 2023-10-15 at 12 59 59

**Review by gund — APPROVED** [review]
This looks great! Thanks for your effort!

File: README.md (hunk 2 above)
Comment (gund) [reply README.md]: Thanks for linking this one, it could be a nice addition actually, I'm always in for automating stuff.

**Comment by karlhorky** [comment]
Glad to help, thanks for the review and merge!
